src/debug.cc

```diff
@@ -210,38 +272,29 @@ static void
 debugOpenLog(const char *logfile)
 {
     if (logfile == NULL) {
-        debug_log = stderr;
+        TheLog.clear();
         return;
     }
 
-    if (debug_log_file)
-        xfree(debug_log_file);
-
-    debug_log_file = xstrdup(logfile);  /* keep a static copy */
-
-    if (debug_log && debug_log != stderr)
-        fclose(debug_log);
-
     // Bug 4423: ignore the stdio: logging module name if present
     const char *logfilename;
     if (strncmp(logfile, "stdio:",6) == 0)
         logfilename = logfile + 6;
     else
         logfilename = logfile;
 
-    debug_log = fopen(logfilename, "a+");
-
-    if (!debug_log) {
+    if (auto log = fopen(logfilename, "a+")) {
+#if _SQUID_WINDOWS_
+        setmode(fileno(log), O_TEXT);
+#endif
+        TheLog.reset(log, logfilename);
+    } else {
         fprintf(stderr, "WARNING: Cannot write log file: %s\n", logfile);
         perror(logfile);
         fprintf(stderr, "         messages will be sent to 'stderr'.\n");
         fflush(stderr);
-        debug_log = stderr;
+        TheLog.clear();
```

Comment:
We are in agreement that the code should be changed to improve reporting of log opening errors.

The changes you sketched do not work well. For example, in a common case of never-opened TheLog, they record the same error to the same stderr stream twice (without reporting the actual `fopen()` call error even once). Also, they rely on debugs() internals working correctly at debugOpenLog() call time. Those internals are not 100% ready for such "early" debugs() calls -- I see minor problems (e.g., a late tz_set() call), but there could be bigger problems as well (e.g., a late openlog() call). I am sure all these problems can be resolved or worked around, but properly resolving them is a lot of work, and I do not want to become responsible for introducing _new_ problems into this code while cutting corners to keep this PR focused.

The correct changes to improve this code are both too complex and unrelated to this particular PR goals (i.e., ensure reporting of very early errors). Hence, they are out of scope IMO.